File: order/src/main/java/com/apapedia/order/model/Order.java
package com.apapedia.order.model;


import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.Getter;

import java.time.LocalDateTime;
import java.util.UUID;
import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "order_table")

public class Order {
    @Id
    private UUID orderId = UUID.randomUUID();

    @NotNull
    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt = LocalDateTime.now();

    @NotNull
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt = LocalDateTime.now();

    @NotNull
    @Column(name = "status", nullable = false)
    private int status = 0;

    @NotNull
    @Column(name = "total_price", nullable = false)
    private int totalPrice = 0;

    @NotNull
    @Column(name = "customer_id", nullable = false)
    private UUID customerId;

    @NotNull
    @Column(name = "seller_id", nullable = false)
    private UUID sellerId;

    @OneToMany(mappedBy = "order", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<OrderItem> listOrderItem;
}

File: mobileapp/lib/app/data/models/order.dart
import 'package:mobileapp/app/data/models/order_item.dart';

class Order {
  final String id;
  final String customerId;
  final String sellerId;
  final int totalPrice;
  final int status;
  final List<OrderItem> listOrderItem;

  Order({
    required this.id,
    required this.customerId,
    required this.sellerId,
    required this.totalPrice,
    required this.status,
    required this.listOrderItem,
  });

  factory Order.fromJson(Map<String, dynamic> json) {
    return Order(
      id: json['orderId'],
      customerId: json['customerId'],
      sellerId: json['sellerId'],
      totalPrice: json['totalPrice'],
      status: json['status'],
      listOrderItem: json['listOrderItem'],
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'customerId': customerId,
        'sellerId': sellerId,
        'totalPrice': totalPrice,
        'status': status,
        'listOrderItem': listOrderItem,
      };
}
